src/objects/zcl_abapgit_object_aqsg.clas.abap

```diff
@@ -0,0 +1,113 @@
+CLASS zcl_abapgit_object_aqsg DEFINITION
+  PUBLIC
+  INHERITING FROM zcl_abapgit_objects_super
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+
+    INTERFACES zif_abapgit_object .
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+
+    METHODS get_generic
+      RETURNING
+        VALUE(ro_generic) TYPE REF TO zcl_abapgit_objects_generic
+      RAISING
+        zcx_abapgit_exception .
+
+    METHODS get_field_rules
+      RETURNING
+        VALUE(ro_result) TYPE REF TO zif_abapgit_field_rules.
+ENDCLASS.
+
+
+
+CLASS ZCL_ABAPGIT_OBJECT_AQSG IMPLEMENTATION.
+
+
+  METHOD get_field_rules.
+
+    ro_result = zcl_abapgit_field_rules=>create( ).
+
+* add rules here if needed
+
+  ENDMETHOD.
+
+
+  METHOD get_generic.
+    " transaction SQ02
+    CREATE OBJECT ro_generic
+      EXPORTING
+        is_item        = ms_item
+        io_field_rules = get_field_rules( )
+        iv_language    = mv_language.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~changed_by.
+    rv_user = zcl_abapgit_objects_super=>c_user_unknown.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    set_default_transport( iv_transport ).
+
+    get_generic( )->delete( iv_package ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~deserialize.
+
+    set_default_transport( iv_transport ).
+
+    get_generic( )->deserialize(
+      iv_package = iv_package
+      io_xml     = io_xml ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~exists.
+
+    rv_bool = get_generic( )->exists( ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_comparator.
+    RETURN.
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~get_deserialize_steps.
+    APPEND zif_abapgit_object=>gc_step_id-late TO rt_steps.
```

Comment:
any reason why this is not `abap`?

Reply:
I dont think anything can depend on ABAP query stuff, so moving it to late. I think ABAP Query can reuse ABAP code, but not sure

"Used for objects that depend on other objects processed in the previous two phases"

Reply:
👍